doc/scripts/gen_state_diagram.py

```diff
@@ -73,6 +73,17 @@ def __init__(self):
         self.subgraphnum = 0
         self.clusterlabel = {}
 
+        self.color_idx = 0
+        self.color_palette = [
+            "#000000",  # black
+            "#0000ff",  # blue
+            "#ffa500",  # orange
+            "#00ff00",  # lime
```

Comment:
nit, lime can be difficult to read on a white background. how about using `#00b300` or `#6cc644`?

Reply:
Thanks for your feedback. I found that one existing color #66cdaa closes to the two colors you proposed. So I decide to switch to another color set that doesn't have the lime. I chose these colors based on this website: https://mokole.com/palette.html, with the following parameters but skipping the lime color from the result:
![](https://user-images.githubusercontent.com/468515/99621866-c5531600-29dd-11eb-8b1b-51f565c60c07.png)

This is how the resulting diagram looks:
![peering_graph generated](https://user-images.githubusercontent.com/468515/99622033-2c70ca80-29de-11eb-8aa7-8647b2170af7.png)